### DISCUSSION TITLE
Version display problem on Esp32-c3 SuperMini after OTA update

### DISCUSSION THREAD

**Discussion by SuperNinja73** [tbnobody/OpenDTU#1698] [Q&A]
If I try to flash my Esp32-c3 SuperMini ![IMG20240127192727](https://github.com/tbnobody/OpenDTU/assets/53425740/c53783d5-6d1f-4600-a9fe-fb68edfb6e4e)
with a flash tool and opendtu-generic_esp32.factory.bin the ESP reboots without stopping :  
```
invalid header: 0xffffffff
invalid header: 0xESP-ROM:esp32c3-api1-20210207
Build:Feb  7 2021
rst:0x15 (USB_UART_CHIP_RESET),boot:0xf (SPI_FAST_FLASH_BOOT)
Saved PC:0x4004c0d4
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
```

So I use Platformio with the Source code (zip) (23.11.16) and upload the "generic_esp32c3" and I manage to connect to the WEB UI (everything is fine).
![try update 23 11 16  to 24 1 26](https://github.com/tbnobody/OpenDTU/assets/53425740/3a600a83-1841-4f1f-9d3c-869f22d04ef6)
I download the latest version of the Source code (zip) (24.1.26) and run the Build to obtain the firmware.bin in .pio folder
I open the WEB ui and launch the firmware upgrade OTA with firmware.bin, which is carried out correctly **but I display a version "00000"**
![24 1 26](https://github.com/tbnobody/OpenDTU/assets/53425740/5e080080-9b14-49c8-9380-3d492f6eb9b2)
Anything to change in the Source code? Or a Bug?

**Comment by MetaChuh** [comment]
do not download the source as a zip file.
you have to checkout/clone your desired opendtu revision from within your platformio ide or the command line.

see the step by step guides for [visual studio code](https://tbnobody.github.io/OpenDTU-docs/firmware/compile_vscode/)
and alternatively for [the command line using pio core](https://tbnobody.github.io/OpenDTU-docs/firmware/compile_cli/)

**Comment by SuperNinja73** [comment]
ok so for each new release, I must clone the git repository, do the Buid with VSCode and then, "Upgrade Firmware by OTA" by selecting firmware.bin in .pio ?
I would like to avoid launching VSCode and do a Build for each release. The latest release opendtu-generic_esp32.bin does not integrate my ESP32-c3 ? It seems to me that it is supported?

**Reply by MetaChuh** [reply]
no, you only have to clone the repo once.

just click on the version at the left bottom
(shows "master" if you just did a fresh clone)
and then select the desired older, newer, or future version to build

here's a screenshot for better explanation:

<img width="681" alt="Screenshot OpenDTU Checkout 2024-01-29 - 21 58 57" src="https://github.com/tbnobody/OpenDTU/assets/45773305/34ce1237-f183-400a-b031-0d9975bc3f73">

note: the build environment on the screenshot should also be limited to c3, to skip unnecessary builds for other platforms.

ps: i know, seems like a little inconvenience, as the c3 is currently not within the release build script.
but currently it's even less clicks, than opening a browser, github, search and download the latest bin (if you have vsc configured to always open the last state and projects.

**Reply by SuperNinja73** [reply]
It’s great, I like it! So if I stay on the master version it automatically builds the latest release, limiting only to the environment I chose: c3.
 If a new version is present on Git, does it download it automatically? Or should I tell it to check and synchronize with the remote Git? With this button perhaps?
![image](https://github.com/tbnobody/OpenDTU/assets/53425740/b404114b-cd6c-4b46-8322-4939000dc11c)
